sia/miner/mine.go

```diff
@@ -54,7 +54,14 @@ func (m *Miner) mine() {
 
 		// If we are allowed to be running, mine a block, otherwise shut down.
 		if desiredThreads >= myThread {
-			m.SolveBlock()
+			// Grab the necessary variables for mining, and then attempt to
+			// mine a block.
+			m.mu.RLock()
+			bfw := m.blockForWork()
+			target := m.target
+			iterations := m.iterationsPerAttempt
+			m.mu.RUnlock()
+			m.solveBlock(bfw, target, iterations)
```

Comment:
This is kind of gross, but it follows conventions to do it this way. Otherwise, solveBlock needs to grab mutexes, which violates convention.

There really wasn't a pretty way to do this. Channels might clean things up, but I'm not really worried about it.